### ISSUE TITLE
How to initiate a disconnection?

### ISSUE THREAD

**Issue by adamradocz** [davidfowl/BedrockFramework#127]
How to disconnect the client from the server?

**Comment by KieranDevvs** [comment]
Im currently using `ConnectionContext.Abort();`

**Comment by adamradocz** [comment]
Yes, I tried that one as well. The problem with that is, when I tested the `ConnectionContext.Abort();` the server didn't receive the `IsCompleted` nor the `IsCancelled` result. You can check it for yourself, if you add logging at the start and at the end of the `OnConnectedAsync()` method.

Though when I explicitly called the `await connection.DisposeAsync();`, the server received the `IsCompleted` result.

I created a PR for that: #130

**Comment by KieranDevvs** [comment]
Shouldnt `BaseConnectionContext` also have a way of getting the connection state i.e if its closed, disconnected or connected?
Im currently using the cancellation token to see if the context should still be read from
`public virtual CancellationToken ConnectionClosed`

```cs
while (!connectionContext.ConnectionClosed.IsCancellationRequested)
{
    await connectionSession.Reader.ReadAsync(_protocol);
    var packet = result.Message;

    //.... Handle packet ....
}
```

**Comment by adamradocz** [comment]
@davidfowl In  #130 PR. In the samples/ClientApplication/Program.cs  at line 366, if I replace the `await connection.DisposeAsync();` for `connection.Abort();` nothing happens. The server won't receive `IsCompleted` nor `IsCancelled` result.

**Comment by davidfowl** [comment]
Calling DisposeAsync to abort is incorrect. Calling Abort should work, if it doesn't work then there's a bug. DisposeAsync is implicitly called by the pipeline when the code unwinds.
